pkg/kapis/devops/v1alpha3/handler.go

```diff
@@ -415,15 +415,20 @@ func (h *devopsHandler) DeleteCredential(request *restful.Request, response *res
 	}
 }
 
-func (h *devopsHandler) getDevOps(request *restful.Request) (devops.DevopsOperator, error) {
+func (h *devopsHandler) getDevOps(request *restful.Request) (operator devops.DevopsOperator, err error) {
 	ctx := request.Request.Context()
 	token := ctx.Value(constants.K8SToken).(string)
 
-	klog.V(9).Infof("get DevOps client with  token: %s", token)
-	kubernetesClient, err := k8s.NewKubernetesClientWithToken(token, h.k8sClient.Config().Host)
-	if err != nil {
-		return nil, err
+	var k8sClient k8s.Client
+	if token == "" {
+		k8sClient = h.k8sClient
```

Comment:
Please explain why we allow people to access our resources without token.

Reply:
as you can see from kubesphere/kubesphere#4129

it does not provide the K8S token. And other modules have the same pattern. Such as, notification manager.